service/src/main/java/com/codeforcommunity/processor/ProtectedUserProcessorImpl.java

```diff
@@ -174,7 +175,14 @@ public void changePrivilegeLevel(
         || userData.getPrivilegeLevel() == PrivilegeLevel.SUPER_ADMIN)) {
       throw new AuthException("User does not have the required privilege level");
     }
+
     String targetEmail = changePrivilegeLevelRequest.getTargetUserEmail();
+
+    // check if target user exists
+    if (!db.fetchExists(db.selectFrom(USERS).where(USERS.EMAIL.eq(targetEmail)))) {
+      throw new UserDoesNotExistException(targetEmail);
+    }
+
     UsersRecord user = db.selectFrom(USERS).where(USERS.EMAIL.eq(targetEmail)).fetchOne();
 
     userRecordExistsCheck(user, new UserDoesNotExistException(targetEmail));
```

Comment:
❓ What does this do?

Reply:
You're right, does the same thing. Removed second check as it relies on the db call that was causing the error.